Thu Aug  3 01:05:32 2000  Jeffrey A Law  (law@cygnus.com)

	* flow.c (find_auto_inc): Verify that we've got a REG before
	peeking at its regno.  Fail, don't abort if we can't find
	the increment of the desired register.

--- gcc/flow.c	2000/08/02 04:21:26	1.316
+++ gcc/flow.c	2000/08/03 07:16:39	1.317
@@ -5018,12 +5018,12 @@ find_auto_inc (pbi, x, insn)
   if (GET_CODE (y) != PLUS)
     return;
 
-  if (REGNO (XEXP (y, 0)) == REGNO (addr))
+  if (REG_P (XEXP (y, 0)) && REGNO (XEXP (y, 0)) == REGNO (addr))
     inc_val = XEXP (y, 1);
-  else if (REGNO (XEXP (y, 1)) == REGNO (addr))
+  else if (REG_P (XEXP (y, 1)) && REGNO (XEXP (y, 1)) == REGNO (addr))
     inc_val = XEXP (y, 0);
   else
-    abort ();
+    return;
 
   if (GET_CODE (inc_val) == CONST_INT)
     {
